src/app/plugins.js

```diff
@@ -0,0 +1,130 @@
+import { fromEntries } from '../utils';
+import { CoordinationType } from './constants';
+import {
+  COMPONENT_COORDINATION_TYPES,
+  DEFAULT_COORDINATION_VALUES,
+} from './state/coordination';
+
+export const PLUGIN_VIEW_TYPES_KEY = '__VITESSCE_PLUGIN_VIEW_TYPES__';
+export const PLUGIN_COORDINATION_TYPES_KEY = '__VITESSCE_PLUGIN_COORDINATION_TYPES__';
+export const PLUGIN_COORDINATION_TYPES_PER_VIEW_KEY = '__VITESSCE_PLUGIN_COORDINATION_TYPES_PER_VIEW__';
+export const PLUGIN_FILE_TYPES_KEY = '__VITESSCE_PLUGIN_FILE_TYPES__';
+
+// Reference: https://github.com/higlass/higlass-register/blob/master/src/index.js
+// TODO: rather than the registration functions and storing things on the window object,
+// should plugins be provided as props to the main <Vitessce/> React component
+// and stored via some React context?
+window[PLUGIN_VIEW_TYPES_KEY] = window[PLUGIN_VIEW_TYPES_KEY] || {};
+window[PLUGIN_COORDINATION_TYPES_KEY] = window[PLUGIN_COORDINATION_TYPES_KEY] || {};
+window[PLUGIN_COORDINATION_TYPES_PER_VIEW_KEY] = (
+  window[PLUGIN_COORDINATION_TYPES_PER_VIEW_KEY] || {}
+);
+window[PLUGIN_FILE_TYPES_KEY] = window[PLUGIN_FILE_TYPES_KEY] || {};
+
+/**
+ * Register a new coordination type.
+ * @param {string} typeName Name for the new coordination type.
+ * @param {string[]} supportedViews A list of view types that should support this coordination type.
+ * @param {*} defaultValue A default value for the coordination type.
```

Comment:
Not correct?

Reply:
Done in https://github.com/vitessce/vitessce/pull/1212/commits/8964b41c74fd7782e5c7f7ba901ab8a2966b846b#diff-7ba443b15fd323266f46e5984b4a8dbf45306b45ab6f4b082041c4b4f9fcd222L27